closed/GensrcJ9JCL.gmk

```diff
@@ -47,7 +47,16 @@ ifeq (true,$(OPENJ9_ENABLE_OPENJDK_METHODHANDLES))
   JPP_TAGS += OPENJDK_METHODHANDLES
 endif # OPENJ9_ENABLE_OPENJDK_METHODHANDLES
 
-$(J9JCL_SOURCES_DONEFILE) : $(AllJclSource) $(AllDdrSource)
+DIR_JAVA_BASE_UNIX_JAVA_LANG := src/java.base/unix/classes/java/lang
+FILE_PROCESS_ENVIRONMENT := $(TOPDIR)/$(DIR_JAVA_BASE_UNIX_JAVA_LANG)/ProcessEnvironment.java
+OVERLAY_FILES := $(FILE_PROCESS_ENVIRONMENT)
+addOpenJDKJavaSource : $(OVERLAY_FILES)
+	@$(ECHO) addOpenJDKJavaSource: $(OVERLAY_FILES)
+	@$(MKDIR) -p $(TOPDIR)/openj9/jcl/$(DIR_JAVA_BASE_UNIX_JAVA_LANG)
+	# CRIU only supports x86-64 Linux.
+	@$(CP) -v $(FILE_PROCESS_ENVIRONMENT) $(TOPDIR)/openj9/jcl/$(DIR_JAVA_BASE_UNIX_JAVA_LANG)
```

Comment:
This has some problems:
* `addOpenJDKJavaSource` should be a `.PHONY`
* the `addOpenJDKJavaSource` rule will run unconditionally
* this copies a file into the openj9 working tree, leaving it 'dirty' with the danger that someone will commit `ProcessEnvironment.java` to the openj9 repository
* comments should precede the rule (they don't need to be in the executable script)

Reply:
Added `addOpenJDKJavaSource` to the `.PHONY` target list, removing the file copied after generating J9JCL sources.

Reply:
You haven't addressed the other three issues. Removing the file afterward doesn't avoid polluting the openj9 working tree if the preprocessor fails.

Reply:
> Removing the file afterward doesn't avoid polluting the openj9 working tree if the preprocessor fails.

This means something wrong already, the error should be checked including if there is any unintended change.

`addOpenJDKJavaSource` depends on `$(OVERLAY_FILES)`.
Not sure what other issues.

Reply:
No, the problem I describe was not already present.

My suggestion is to copy `ProcessEnvironment.java` to a new `support` sub-directory and run the preprocessor again using that as the source directory to mix that new contentinto `$(J9JCL_SOURCES_DIR)`. That way nothing is written into the openj9 working tree, not even temporarily.

Reply:
Thanks the suggestion, updated accordingly, pls have another look.